### PR TITLE
Use theme methods instead of macros for application name

### PR DESCRIPTION
**Pull request by erikjv**
And also for the application icon name. There is one left in the common sources, but those cannot depend on libsync, so that's left alone.

### CODE REVIEW

**Comment by TheOneRing** [comment]
I assume there is are a few more. In general you could check to not include config.h which should only be needed in one or two locations the rest should rely on theme.h

**Comment by erikjv** [comment]
Note: `guiutility_mac.mm` is still a problem

**Comment by erikjv** [comment]
> Note: `guiutility_mac.mm` is still a problem

I left two uses of defines in there, as they are mac specific.

**Comment by TheOneRing** [comment]
```
owncloud-client [work/get-app-name-from-theme ≡ +3 ~0 -0 !]> git grep "config.h"
src/common/plugin.cpp:#include "config.h"
src/csync/CMakeLists.txt:configure_file(config.h.in ${CMAKE_CURRENT_BINARY_DIR}/config.h)
src/csync/csync_exclude.cpp:#include "config.h"
src/gui/folderwatcher_linux.cpp:#include "config.h"
src/gui/guiutility_mac.mm:#include "config.h"
src/gui/libcloudproviders/libcloudproviders.cpp:#include "config.h"
src/gui/lockwatcher.h:#include "config.h"
src/gui/main.cpp:#include "csync/config.h"
src/gui/socketapi/socketapi.h:#include "config.h"
src/gui/updater/updater.cpp:#include "config.h"
src/libsync/configfile.cpp:#include "config.h"
src/libsync/filesystem.h:#include "config.h"
src/libsync/theme.cpp:#include "config.h"
test/testsyncvirtualfiles.cpp:#include "config.h"
test/testutility.cpp:#include "config.h"

File: src/gui/systray.cpp [hunk 1]
```diff
@@ -75,8 +74,8 @@ void Systray::showMessage(const QString &title, const QString &message, const QI
 #else
 #ifdef USE_FDO_NOTIFICATIONS
     if (QDBusInterface(NOTIFICATIONS_SERVICE_C(), NOTIFICATIONS_PATH_C(), NOTIFICATIONS_IFACE_C()).isValid()) {
-        QList<QVariant> args = QList<QVariant>() << Theme::instance()->appNameGUI() << quint32(0) << QStringLiteral(APPLICATION_ICON_NAME)
-                                                 << title << message << QStringList() << QVariantMap() << qint32(-1);
+        QList<QVariant> args = QList<QVariant>() << Theme::instance()->appNameGUI() << quint32(0) << Theme::instance()->applicationIcon() << title << message
```
Comment (TheOneRing) [inline src/gui/systray.cpp:77]: This used to be a QString, we now send a QIcon, I don't think that will work.

**Review by TheOneRing — CHANGES_REQUESTED** [review]

**Comment by erikjv** [comment]
This fixes https://github.com/owncloud/client/issues/10461

**Review by TheOneRing — APPROVED** [review]

**Pull request merged**